src/syntax/abs/mod.rs

```diff
@@ -1,6 +1,125 @@
-use crate::syntax::common::Level;
+use crate::check::monad::error::TCE;
+use crate::check::monad::TCM;
+use crate::syntax::common::{DtKind, Level, DBI};
+use crate::syntax::env::NamedEnv_;
+use crate::syntax::surf::ast::{Decl, DeclKind, Expr, Ident};
+use either::Either;
+use std::collections::BTreeMap;
 
 #[derive(Debug, Clone)]
 pub enum Abstract {
-    Type(Level),
+    Type(Ident, Level),
```

Comment:
How about using `SyntaxInfo` instead of `Ident` since `Ident` means an identifier where `SyntaxInfo` represent arbitrary syntactical information?